### ISSUE TITLE
High quality profile picture broken with new API change

### ISSUE THREAD

**Issue by psylenced** [instaloader/instaloader#306]
When running instaloader with logged in setting, there seems to be an issue getting high quality profile pictures.

This occurs on on Win10 with 4.2.7.

This is possibly due to the recent API changes.

```'query_hash' Unable to fetch high quality profile pic.
'query_hash' Unable to fetch high quality profile pic.
'query_hash' Unable to fetch high quality profile pic.
'query_hash' Unable to fetch high quality profile pic.
```

**Comment by Showmaster3000** [comment]
I cannot confirm that. Using WIn7 with 4.2.7 since yesterday and new profile pics of checked accounts are in 1080 here.
Can you please post your commandline to reproduce your problem?

**Comment by psylenced** [comment]
Hi,

Command line is following:

```instaloader --no-captions --no-metadata-json --login xxxx --sessionfile session.txt --fast-update --filename-pattern="{profile}_{date}_{owner_id}-{mediaid}" +all.txt ```

**Comment by aandergr** [comment]
I cannot confirm that neither. Could you show us an example of where this happens?

**Comment by psylenced** [comment]
I looked back through the logs, and it appears to be related to 429 (too many requests), also possibly a false error.

```
[  1/250] [USER1]/[USER1]_2019-05-21_10-10-21_11111-20481111111111111111.jpg exists
JSON Query to api/v1/users/1111111/info/: 429 Too Many Requests [retrying; skip with ^C]
'query_hash' Unable to fetch high quality profile pic.

[USER2]/2014-10-30_05-26-50_UTC_profile_pic.jpg already exists
Retrieving posts from profile [USER2].
[  1/ 12] [USER2]/[USER2]_2016-07-21_06-23-55_2222-12992222222222.jpg exists
JSON Query to api/v1/users/22222222/info/: 429 Too Many Requests [retrying; skip with ^C]
'query_hash' Unable to fetch high quality profile pic.

[USER3]/2019-04-28_01-51-30_UTC_profile_pic.jpg already exists
Retrieving posts from profile [USER3].
[  1/ 97] [USER3]/[USER3]_2017-08-04_18-31-01_333333-157433333333333.jpg exists
```

When I run through on the individual profiles I get no error. Also they don't seem to have new profile pictures. So when I run the above 3 users it works fine.

```
Loaded session from session.txt.
Logged in as xxxxxx.
Downloading 3 profiles: [USER1] [USER2] [USER3]
[USER1]/2017-06-29_02-47-57_UTC_profile_pic.jpg already exists
[USER2]/2014-10-30_05-26-50_UTC_profile_pic.jpg already exists
[USER3]/2019-04-28_01-51-30_UTC_profile_pic.jpg already exists
Saved session to session.txt.
```

The full list has a lot more profiles and takes a number of hours.

So at a best guess, if you get a 429 while trying to check the profile pic on this API call: ```api/v1/users/22222222/info/```, you'll get the query_hash error.

I've only started getting the error over the last 2 weeks - so possibly the API response is slightly different too.

**Comment by aandergr** [comment]
Thanks, your complete log helped us to find and fix this bug. The fix is included in the new version 4.2.8.
